### ISSUE TITLE
Add support for multiple NTRIP servers

### ISSUE THREAD

**Issue by nseidle** [sparkfun/SparkFun_RTK_Firmware#767]
The ability to communicate with multiple NTRIP servers is a good thing.

[This PR](https://github.com/sparkfun/SparkFun_RTK_Firmware/pull/766/files) implements most of the work. The serial UI and web config UI need fixing.

**Comment by amlago** [comment]
Very good idea. Is there an implementation date? I am interested in sending data to Rtk2go and emlid at the same time.

**Comment by nseidle** [comment]
Currently two NTRIP Server targets are supported.

**Comment by PaulZC** [comment]
Hi Angel (@amlago ),

This feature is in the June 10th Release Candidate firmware. Please give it a try. You can enter the credentials for the two NTRIP servers either through the serial menus or the web configuration page. I have been testing it with RTK2go and emlic caster and it works well.

For the serial menu:

Enter 3 to open the Base Menu
Enter 7 to enable the NTRIP Server(s)
For Server 1, you can enter the credentials with 10-14
Enter 8 to toggle to Server 2
Then enter the credentials for Server 2 using 10-14

![image](https://github.com/sparkfun/SparkFun_RTK_Firmware/assets/5690545/75693b99-5553-4921-9b29-30e91de06b38)

The firmware will only attempt to connect to a server if the Caster Host, Port and Mountpoint are defined.

Best wishes,
Paul

**Comment by amlago** [comment]
I will try it and make the corresponding comments.
My greetings
Angel

**Comment by amlago** [comment]
Hello, I have tested version 4.1 from 10/06.
Regarding issuing data to 2 casters, it has been a success.
I see fixes from rtk2go and from emlid.
Now the problem is that if I use Config wifi and I want to use only one caster and leave the other empty, I can't. It won't let me save the changes.
There should be a check in front of each caster to use it or not.
If I leave the 2nd caster lockers empty I get an error when saving.
On the other hand, entering the caster password is not a mandatory requirement for rtk2go to receive and issue data. They only require the user's email to authenticate. That's what I understand and until yesterday it worked for me with version 3.8. I never entered the caster password. I don't mind entering it, I'm just saying that if it is left blank, the wifi config should save the changes the same.
In short, it is a very good advance, we should polish the interface details in the WiFi config.
greetings.

**Comment by PaulZC** [comment]
Hi Angel (@amlago ),

Thank you for the feedback - and apologies for the mistake!

I have updated the WiFi Config page in today's June 14th Release Candidate binary. Now, only the Server 1 Caster Host, Port and Mount Point are mandatory. The Server 2 configuration can be left empty; set the port to 0 if you are not using it.

I made some other small improvements:
* The TCP / UDP settings are now in their own section
* I added the missing TCP Client Host setting
* The Ethernet TCP settings have been deleted - the TCP / UDP settings are common to both WiFi and Ethernet
* The NTRIP Client Host and User validation has been improved: if rtk2go.com is set as the Host, the User must include an email address with an '@' symbol

Please try the new binary when you have time.

Have a nice weekend,
Paul

**Comment by amlago** [comment]
Hello Paul.
I have tested version 4.1 06/14.
The emission of corrections in the 2 casters works very well.
Now if I want to use a single caster, I put 0 in the port of the one I'm not going to use.
It would function as an enabling check.
I have tried that and it works.
So Eureka.
My Facet almost always works in base mode so for me they are very useful functions.
My greetings and have a good weekend. A lot of rain is expected in Uruguay. uffff
Angel

**Comment by nseidle** [comment]
Fixed in [v4.1](https://github.com/sparkfun/SparkFun_RTK_Firmware/releases/tag/v4.1).
